### DISCUSSION TITLE
How can I send a WebRTC live stream that uses VP8 codec?

### DISCUSSION THREAD

**Discussion by tremzo** [ant-media/Ant-Media-Server#5568] [FAQs]
I want to start a WebRTC live stream that uses the VP8 codec and prioritize the VP8 codec for playback when it's available. Is this possible when both H264 and VP8 codecs are checked in the application settings?

**Comment by Mohit-3196** [comment]
Hi @timantmedia ,
Thank you for your great question.

In WebRTC, the choice of codec priority depends on the negotiation process during the WebRTC session establishment between the two peers (typically a browser or application running WebRTC). The priority of codecs is determined by a combination of factors including the capabilities of the endpoints, the supported codecs on both ends and the negotiation process itself, etc.
The codec selected for the session is the one that appears highest in both peers' preference lists and is supported by both peers. 

Ant Media Server supports both H.264 and VP8 codecs for WebRTC and if both are selected, VP8 gets a higher priority for WebRTC when supported by the device.
So yes, if both H.264 and VP8 are enabled, the VP8 gets higher priority.

Thank you

**Reply by mekya** [reply]
Hi @Mohit-3196,

Thank you. 
How to enable H264 and VP8 in Ant Media Server, could you please just let us know some simple screenshot to see where to change?

PS: As far as I remember, Ant Media Server prefers to use H264 over VP8 if both of them are enabled ;)

**Reply by tremzo** [reply]
If I have both VP8 and H264 enabled with both configured to record an MP4 and WebM accordingly, then live streaming WebRTC using Chrome that supports both, I end up only with an MP4 file when I have no ABR selected. 

Is this Chromes behavior using H264 or Ant Media Server preferring to use H264 and recording the MP4?

If I add ABR, then I get both a WebM and MP4 recording.

**Reply by Mohit-3196** [reply]
Thank you @mekya,

> PS: As far as I remember, Ant Media Server prefers to use H264 over VP8 if both of them are enabled

Then it is my bad to have thought the other way.
Thank you very much for the correction 💯  :)

> How to enable H264 and VP8 in Ant Media Server, could you please just let us know some simple screenshot to see where to change?

Yes, sure.
- Just log in to the web panel.
- Go to the Application -> Settings
- Check H.264 and VP8
![Screenshot 2023-09-19 170204](https://github.com/ant-media/Ant-Media-Server/assets/86982446/8b1a1909-b911-4a4a-a797-c449f6f3716a)
If both are enabled, and there is no Adaptive setting, then the server will simply forward the incoming video codec.
Also, when the Adaptive setting is enabled, the server will transcode the incoming stream to make both codecs available.

@timantmedia ,

> Is this Chromes behavior using H264 or Ant Media Server preferring to use H264 and recording the MP4?

If both H.264 and VP8 are enabled and there's no adaptive setting then the recording happens according to the incoming stream video codec. i.e., If the incoming stream codec is H.264 then it will be recorded with MP4, if VP8 then with WebM.

@mekya, please let me know if I misunderstood anything :) 

Thank you.

**Reply by mekya** [reply]
I think you're right. 

This discussion makes me think that it's good to send VP8 and H264 as simulcast by client sides if both of them are enabled so user will have more compatibility and less process overhead on the server side.
